### PR TITLE
feat: add auto-release

### PR DESCRIPTION
**Pull request by Steve0x2a**
Signed-off-by: Steve0x2a <stevesough@gmail.com>

### CODE REVIEW

**Comment by hsluoyz** [comment]
@ffyuanda plz review

File: .github/build.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+          python-version: 3.8
+            
+      - name: Install setuptools
+        run: python -m pip install --upgrade setuptools wheel twine
+      
+      - name: Release
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          PYPI_TOKEN: ${{ secrets.PYPI_TOKEN }}
+        run: npx semantic-release
```
Comment (ffyuanda) [inline .github/build.yml:39]: Better leave a blank line at the end.

File: .releaserc.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      ],
+        [
+        "@semantic-release/git",
+        {
+          "message": "chore(release): ${nextRelease.version} [skip ci]\n\n${nextRelease.notes}",
+          "assets": ["CHANGELOG.md", "setup.py", "setup.cfg"]
+        }
+      ]
+    ]
+  }
```
Comment (ffyuanda) [inline .releaserc.json:23]: ditto

File: setup.cfg [hunk 3]
```diff
@@ -37,4 +37,4 @@ test_suite = tests
 universal = true
 
 [sdist]
-formats = zip, gztar
+formats = gztar
```
Comment (ffyuanda) [inline setup.cfg:40]: Are we only leaving tarballs?

File: setup.py [hunk 4]
```diff
@@ -0,0 +1,4 @@
+
+from setuptools import setup, find_packages
+
+setup()
```
Comment (ffyuanda) [inline setup.py:4]: ditto

**Review by ffyuanda — CHANGES_REQUESTED** [review]
Please leave a blank line at the end of every file.

**Comment by ffyuanda** [comment]
@Steve0x2a Please also put the [test.pypi.org](https://test.pypi.org/) link here, thanks.

File: setup.cfg (hunk 3 above)
Comment (Steve0x2a) [reply setup.cfg:40]: @ffyuanda If two packages are generated at the same time, using semantic-release to upload to pypi will generate an error. I'm not sure if there is a better way to control uploading only one zip.

**Comment by Steve0x2a** [comment]
> @Steve0x2a Please also put the [test.pypi.org](https://test.pypi.org/) link here, thanks.

@ffyuanda https://test.pypi.org/project/casdoor-python-sdk-test/

File: setup.cfg (hunk 3 above)
Comment (ffyuanda) [reply setup.cfg:40]: Could you please paste the link to the error here? Guess it's somewhere in the Actions CI process?

Comment (Steve0x2a) [reply setup.cfg:40]: https://github.com/Steve0x2a/test-casdoor-python-ci/runs/5491755663?check_suite_focus=true 
image

Comment (ffyuanda) [reply setup.cfg:40]: Alright, we can keep `gztar` as long as it's useful enough.

**Comment by ffyuanda** [comment]
@Steve0x2a According to both [setuptools](https://setuptools.pypa.io/en/latest/userguide/quickstart.html#transitioning-from-setup-py-to-setup-cfg) and [PyPA](https://packaging.python.org/en/latest/tutorials/packaging-projects/#configuring-metadata), we should transition from `setup.py` to `setup.cfg`. Are there any reasons that we still need the `setup.py` file when packaging and distributing?

**Comment by Steve0x2a** [comment]
@ffyuanda This is because the semantic-release-pypi we use, I tried to just use setup.cfg, but it doesn't work normally. I also tried to use semantic-release-python, but it seems that it has a problem parsing our pyproject.xml:https://github.com/Steve0x2a/test-casdoor-python-ci/runs/5491449127?check_suite_focus=true
I thought it might be a good idea to replace an auto-release tool, but I'm not familiar with this.

**Comment by ffyuanda** [comment]
@Steve0x2a @hsluoyz OK, this PR has fulfilled its mission: add auto-release. If we need a more compatible auto-release tool, we should open another PR. As for now, LGTM.

**Review by ffyuanda — APPROVED** [review]
LGTM

**Comment by hsluoyz** [comment]
@ffyuanda @Steve0x2a why the CI doesn't run? Is there anything else we need to do to enable the CI?

https://github.com/casdoor/casdoor-python-sdk/actions

image

**Comment by ffyuanda** [comment]
@hsluoyz If I'm not mistaken, the CI is configured to be run only n to events: push or pull request, see here
https://github.com/casdoor/casdoor-python-sdk/pull/11/files#diff-5c4edb507e52e6e8a1c65945eb84b903c293cdf29ef8194ea316c0dd89d7319eR3-R6

**Comment by ffyuanda** [comment]
@hsluoyz Probably we should use a new PR or push to trigger it.

**Comment by hsluoyz** [comment]
@Steve0x2a can you add a small PR like adding a few texts for README? https://github.com/casdoor/casdoor-python-sdk/issues/7

**Comment by hsluoyz** [comment]
@leeqvip

**Comment by leeqvip** [comment]
@hsluoyz  I think this PR should be reversed. The configuration directory for `Github Actions` should be `.github/workflows`

**Comment by ffyuanda** [comment]
Agree.

**Pull request merged**